doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -424,50 +391,55 @@ automatically created.
          .. coqtop:: reset all
 
             Require Import List.
-            Hint Extern 5 ({?X1 = ?X2} + {?X1 <> ?X2}) => generalize  X1, X2; decide equality : eqdec.
+            Hint Extern 5 ({?X1 = ?X2} + {?X1 <> ?X2}) => generalize  X1, X2;
+              decide equality : eqdec.
             Goal forall a b:list (nat * nat), {a = b} + {a <> b}.
             Info 1 auto with eqdec.
 
-   .. cmdv:: Hint Cut @regexp : @ident
-      :name: Hint Cut
+   .. cmd:: Hint Cut [ @hints_regexp ] {? : {+ @ident } }
+
+      .. insertprodn hints_regexp hints_regexp
+
+      .. prodn::
+         hints_regexp ::= {+ @qualid }
+         | _
+         | @hints_regexp %| @hints_regexp
+         | @hints_regexp @hints_regexp
+         | @hints_regexp *
+         | emp
+         | eps
+         | ( @hints_regexp )
+
+      The `emp` regexp does not match any search path while `eps`
+      matches the empty path.
+
+      Used to cut the proof search tree according to a regular
+      expression that matches the paths to be cut. Note that there is no
+      operator precedence during parsing, one can
+      check with :cmd:`Print HintDb` to verify the current cut expression:
```

Comment:
To me this description is completely out of order. For instance, the first sentence of the second paragraph should come first. The third paragraph (currently after the warning) should be moved up, closer to the initial description. The second sentence of the second paragraph belongs to a warning box.

Reply:
I think I've improved but I don't fully understand it (what exactly is cut, how is @ident a hint by itself, what is the current path).  OTOH I don't think I've lost any info that was there.  Please take a look.

Reply:
Is there a way to print the current cut expression and the current path?

Reply:
See this artificial example:

```coq
Print HintDb typeclass_instances.
(* Cut: emp *)

Class foo := {}.

Set Typeclasses Debug.

Goal foo.
Fail typeclasses eauto.
(*
1: looking for foo without backtracking
1: no match for foo, 0 possibilities
*)
Abort.

Instance bar : foo := {}.

Goal foo.
typeclasses eauto.
(*
1: looking for foo without backtracking
1.1: exact bar on foo, 0 subgoal(s)
Finished run 1 of resolution.
*)
Qed.

Hint Cut [ bar ] : typeclass_instances.

Print HintDb typeclass_instances.
(* Cut: bar *)

Goal foo.
Fail typeclasses eauto.
(*
1: looking for foo without backtracking
1: no match for foo, 1 possibilities
*)
Abort.
```